PerformanceRefactor.md

```diff
@@ -0,0 +1,41 @@
+# Performance Refactor Proposal
+
+This document describes the proposed refactoring of Translator at the start of the Performance phase at a high level. This does not describe all details of the refactoring, which will be jointly developed over time through discussions and decisions in the relevant working groups.
+The goal of this document is to establish a broad agreed-upon framework into which those details will fit, and codify the discussions at the February 2025 Relay.
+
+The plan to to complete the refactoring by Feb 28, 2026.
+
+## ARAs to Shepherd
+
+* We will create a shared platform for ARA implementation
+* The name of this platform is Shepherd
+* Shepherd will be implemented in python
+* Shepherd will be implemented as a workflow engine with shared operation components
+* Technical decisions related to the implementation of workflows are explicitly outside the scope of this document
+* ARAs will maintain individual TRAPI endpoints addressable at individual URLs, even if the underlying implementation is shared.
+* Best effort will be made to reimplement ARAGORN, BTE, and ARAX to Shepherd.
+* ARS will continue to exist and does not require modifications based on the outlined plan for Shepherd.
+* It is possible to host non-Shepherd ARAs if desired and this PR does not exclude the possiblity.
+* Shepherd will access Translator knowledge providers via the Retriever interface.
```

Comment:
This is something that David K brought up, but Shepherd will default to using the Retriever interface for any lookups (and we will strongly recommend this interface), but it's entirely possible that an ARA is able to access other knowledge providers or services outside of Retriever.

Reply:
Thanks @maximusunc I added a new line that ARAs can access their own internal stuff like databases.  @dkoslicki do you think that this captures it or should it be more expansive?

Reply:
The databases part makes sense. And the `It is possible to host non-Shepherd ARAs if desired and this PR does not exclude the possibility.` also leaves open the possibility of using an ARAX-style beam-search for lookups instead of retriever directly, like we talked about on the call. So both of these capture it in my mind.

Reply:
It's probably mostly out of scope of this PR, but I'm still hopeful that Retriever will be fast enough to facilitate ARAX-style beam-search. Or is there something we've discussed that you think will make that impossible or highly unlikely, @dkoslicki ?